### PR TITLE
Allow duplicate step classes inside a pipeline

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes
This PR allows using the same step class multiple time as long as as the step instances have different names.
This allows us to remove the `clone_step` function which was causing troubles in notebooks as well as containerized executions under certain conditions.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [x] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/steps/base_step.py [hunk 1]
```diff
@@ -744,7 +746,7 @@ def configure(
         settings: Optional[Mapping[str, "SettingsOrDict"]] = None,
         extra: Optional[Dict[str, Any]] = None,
         merge: bool = True,
-    ) -> None:
+    ) -> T:
```
Comment (stefannica) [inline src/zenml/steps/base_step.py:749]: How did you get away with not adding a `Returns:` section to your docstring here ?

**Review by stefannica — APPROVED** [review]
Stellar approach !

File: src/zenml/steps/base_step.py (hunk 1 above)
Comment (stefannica) [reply src/zenml/steps/base_step.py:749]: Nevemind, outdated comment.

Comment (schustmi) [reply src/zenml/steps/base_step.py:749]: I used my special ability of ignoring the linting script locally and pushing anyways :P

**Pull request merged**